proxy/http/HttpTransact.cc

```diff
@@ -2683,13 +2683,10 @@ HttpTransact::HandleCacheOpenReadHit(State *s)
       }
     }
 
-    if (server_up || s->stale_icp_lookup) {
+    if (server_up && s->stale_icp_lookup == false) {
```

Comment:
I don't think this is correct, because now line 2721 is always false and 2722 is unreachable code.